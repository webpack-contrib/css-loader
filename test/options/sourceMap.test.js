/* eslint-disable
  prefer-destructuring,
  no-underscore-dangle,
*/
import path from 'path';
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('sourceMap', () => {
    test('{Boolean}', async () => {
      const config = {
        loader: {
          test: /\.css$/,
          options: {
            sourceMap: true,
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { map } = stats.compilation.modules[1]._source.sourceAndMap();

      // Strip host specific paths for CI
      map.sources = map.sources
        .map(source => path.relative(__dirname, source));

      expect(map).toMatchSnapshot();
    });
  });
});
