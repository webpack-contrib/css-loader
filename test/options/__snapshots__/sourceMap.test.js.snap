// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options sourceMap {Boolean} 1`] = `
Object {
  "file": "x",
  "mappings": "AAEA;EACE,qCAAwC;CACzC;;AAED;EACE,YAAY;AACb",
  "names": Array [],
  "sources": Array [
    "../../fixture.css",
  ],
  "sourcesContent": Array [
    "@import './import/import.css';

.url {
  background-image: url('./urls/file.png')
}

.fixture {
  width: 100%;
}
",
  ],
  "version": 3,
}
`;
